### PR TITLE
Documentation for manually implemented mapping methods

### PR DESCRIPTION
**Pull request by coffman21**
Please see https://github.com/mapstruct/mapstruct.org/issues/101

### CODE REVIEW

File: documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc [hunk 1]
```diff
@@ -253,6 +253,61 @@ This means that it is possible for MapStruct not to report unmapped target prope
 ====
 
 
+[[invoking-custom-mapping-method]]
+=== Invoking custom mapping method
+
+Sometimes mappings are not straightforward, and some fields should be mapped using plain java code, while mapping latter of them is covered by generated method.
```
Comment (sjaakd) [inline documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc]: while mapping latter == while mapping the others

File: documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc [hunk 2]
```diff
@@ -253,6 +253,61 @@ This means that it is possible for MapStruct not to report unmapped target prope
...
+
+Let's see if there is a properties `length`, `width` and `height` in `FishTank`, and `VolumeDto` inner object, which contains `volume` and `description` fields in `FishTankWithVolumeDto`.
+We could convert rest of the fields using mappings defined inside `@Mapper` annotations, however it requires implementing method to create `VolumeDto`.
```
Comment (sjaakd) [inline documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc]: rest == the remainder

Comment (sjaakd) [inline documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc]: defined inside `@Mapper` == defined by means of `@Mapping` annotations

File: documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc [hunk 3]
```diff
@@ -253,6 +253,61 @@ This means that it is possible for MapStruct not to report unmapped target prope
...
+
+@Mapper
+public class FishTankMapperWithVolume {
+
+    @Mapping(target = "fish.kind", source = "fish.type")
+    @Mapping(target = "material.materialType", source = "material")
+    @Mapping(target = "quality.document", source = "quality.report")
+    FishTankWithVolumeDto map( FishTank source );
+
+    VolumeDto mapVolume(FishTank source) {
```
Comment (sjaakd) [inline documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc]: Have you tried the example?

**Comment by sjaakd** [comment]
@coffman21 .. Can I ask you (since you are working in the same area in the doc) to also fix https://github.com/mapstruct/mapstruct/issues/1995 ? It's just a minor thing and don't want to confront you with a merge :smile:

File: documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc [hunk 4]
```diff
@@ -253,6 +253,61 @@ This means that it is possible for MapStruct not to report unmapped target prope
...
+    VolumeDto volume;
+}
+
+public class VolumeDto {
+    int volume;
+    String description;
+}
+
+@Mapper
+public class FishTankMapperWithVolume {
```
[the remark this answers is not in the document]
Comment (coffman21) [reply documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc]: Fixed

File: documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc (hunk 1 above)
Comment (coffman21) [reply documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc]: Fixed

File: documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc (hunk 2 above)
Comment (coffman21) [reply documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc]: Fixed

Comment (coffman21) [reply documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc]: Fixed

File: documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc (hunk 3 above)
Comment (coffman21) [reply documentation/src/main/asciidoc/chapter-5-data-type-conversions.asciidoc]: My bad, I haven't tried it before now. I fixed some mistakes in code.

**Comment by coffman21** [comment]
@sjaakd  
> Can I ask you (since you are working in the same area in the doc) to also fix #1995 ?  
Sure thing; should I create a new branch for that or is it ok to fix it in this PR?

**Comment by sjaakd** [comment]
Thanks a lot! Just add a commit to your PR

**Comment by coffman21** [comment]
@sjaakd Done!

**Comment by sjaakd** [comment]
@coffman21 thanks for your input. I did adapt the text a bit more (hopefully for the better, just have a look). Also I added `source.xyz` to the other source fields (this should work and is a bit more explicit).  Have a look here: https://github.com/mapstruct/mapstruct/commit/57cfd35afc194aec5b4130acd3e57f273aa5b68a

**Comment by coffman21** [comment]
@sjaakd Great, glad I helped!

**Pull request closed without merging**